scala-package/examples/src/test/scala/org/apache/mxnetexamples/rnn/ExampleRNNSuite.scala

```diff
@@ -43,37 +44,43 @@ class ExampleRNNSuite extends FunSuite with BeforeAndAfterAll {
   }
 
   test("Example CI: Test LSTM Bucketing") {
-    val tempDirPath = System.getProperty("java.io.tmpdir")
-    var ctx = Context.cpu()
-    if (System.getenv().containsKey("SCALA_TEST_ON_GPU") &&
-      System.getenv("SCALA_TEST_ON_GPU").toInt == 1) {
-      ctx = Context.gpu()
-    }
-    if (!System.getenv().containsKey("CI")) {
-      LstmBucketing.runTraining(tempDirPath + "/RNN/sherlockholmes.train.txt",
-                                tempDirPath + "/RNN/sherlockholmes.valid.txt", Array(ctx), 1)
-    } else {
-      logger.info("Skipping test on CI...")
+    ResourceScope.using() {
+      val tempDirPath = System.getProperty("java.io.tmpdir")
+      var ctx = Context.cpu()
+      if (System.getenv().containsKey("SCALA_TEST_ON_GPU") &&
+        System.getenv("SCALA_TEST_ON_GPU").toInt == 1) {
+        ctx = Context.gpu()
+      }
+      if (!System.getenv().containsKey("CI")) {
+        LstmBucketing.runTraining(tempDirPath + "/RNN/sherlockholmes.train.txt",
+                                  tempDirPath + "/RNN/sherlockholmes.valid.txt", Array(ctx), 1)
+      } else {
+        logger.info("Skipping test on CI...")
+      }
     }
   }
 
   test("Example CI: Test TrainCharRNN") {
-    val tempDirPath = System.getProperty("java.io.tmpdir")
-    if (System.getenv().containsKey("SCALA_TEST_ON_GPU") &&
-          System.getenv("SCALA_TEST_ON_GPU").toInt == 1 &&
-          !System.getenv().containsKey("CI")) {
-      val ctx = Context.gpu()
-      TrainCharRnn.runTrainCharRnn(tempDirPath + "/RNN/obama.txt",
-        tempDirPath, ctx, 1)
-    } else {
-      logger.info("CPU not supported for this test, skipped...")
+    ResourceScope.using() {
```

Comment:
Actually, memory collection happened in the method. You should change that one as well